homeassistant/components/sensor/__init__.py

```diff
@@ -547,6 +547,12 @@ def unit_of_measurement(self) -> str | None:
             unit_of_measurement
             := self.platform.default_language_platform_translations.get(translation_key)
         ):
+            if native_unit_of_measurement is not None:
+                raise ValueError(
+                    f"Sensor {type(self)} has a translation key for "
```

Comment:
We should make sure the integration name (`Entity.platform.name`) is included in this error message.